auth.go

```diff
@@ -53,6 +56,7 @@ type slackAuth struct {
 	auths        chan *slack.OAuthResponse
 	callback     func(*slack.OAuthResponse)
 	api          slackAPI
+	httpHandlers map[string]HTTPHandler
```

Comment:
there is no need to implement a router by hand, you have https://golang.org/pkg/net/http/#ServeMux which is also a HTTP handler

```go
mux := http.NewServeMux()
mux.HandleFunc("/", handler)
mux.HandleFunc("/auth", anotherHandler)
```